[BUG] shfmt failed if comparing arrays

**Describe the bug**
Building fails if there is array comparison.

**To Reproduce**
`echo 'echo [1] < [2]' | amber build - -`

```
 ERROR  Postprocessor 'shfmt' failed
<standard input>:7:17: reached ) without matching $(( with ))
```

**Expected behavior**
Build success.

**Additional context**
version: d288374fe2ecacc33cee36ddee53b86f2e4e58fa

`echo 'echo [1] < [2]' | ./target/release/amber build --no-proc=shfmt --no-proc=bshchk - -`
```bash
#!/usr/bin/env bash
# Written in [Amber](https://amber-lang.com/)
# version: 0.4.0-alpha-22-gd288374
# date: 2025-06-21 00:54:20
__0_array=(1)
__left_comp=("${__0_array[@]}")
__1_array=(2)
__right_comp=("${__1_array[@]}")
__comp="$(
    # Compare if left array < right array
    __len_comp="$((( "${#__left_comp[@]}" < "${#__right_comp[@]}")) && echo "${#__left_comp[@]}" || echo "${#__right_comp[@]}")"
    for (( __i=0; __i<__len_comp; __i++ )); do
        __left="${__left_comp[__i]:-0}"
        __right="${__right_comp[__i]:-0}"
        if (( "${__left}" < "${__right}" )); then
            echo 1
            exit
        elif (( "${__left}" > "${__right}" )); then
            echo 0
            exit
        fi
    done
    (( "${#__left_comp[@]}" < "${#__right_comp[@]}" )) && echo 1 || echo 0
)"
echo "${__comp}"
```

The warned position is

```
__len_comp="$((( "${#__left_comp[@]}" < "${#__right_comp[@]}")) && echo "${#__left_comp[@]}" || echo "${#__right_comp[@]}")"
            ^
            |
            +----- HERE

```

bash-language-server says it is [SC1102](https://www.shellcheck.net/wiki/SC1102)

--- a/src/translate/compare.rs
+++ b/src/translate/compare.rs
@@ -85,9 +85,9 @@ fn create_variable_with_smaller_number(
     right: VarExprFragment
 ) -> (VarStmtFragment, VarExprFragment) {
     let value = SubprocessFragment::new(
-        fragments!("(( ", left.clone().to_frag(), " < ", right.clone().to_frag(), "))",
-            " && echo ", left.clone().to_frag(),
-            " || echo ", right.clone().to_frag())).to_frag();
+        fragments!(" (( ", left.clone().to_frag(), " < ", right.clone().to_frag(), " )) ",
+            "&& echo ", left.clone().to_frag(),
+            "|| echo ", right.clone().to_frag())).to_frag();
     let len_stmt = VarStmtFragment::new(name, Type::Num, value);
     let len_expr = VarExprFragment::from_stmt(&len_stmt).with_render_type(VarRenderType::NameOf);
     (len_stmt, len_expr)

